src/hardware/tandy_sound.cpp

```diff
@@ -631,26 +630,66 @@ static void set_tandy_sound_flag_in_bios(const bool is_enabled)
 	real_writeb(0x40, 0xd4, is_enabled ? 0xff : 0x00);
 }
 
-static void shutdown_dac(Section*)
+static ConfigProfile get_config_profile()
 {
+	switch (machine) {
+	case MCH_PCJR: return ConfigProfile::PCjrSystem; break;
+	case MCH_TANDY: return ConfigProfile::TandySystem; break;
+	default: return ConfigProfile::SoundCardOnly; break;
+	}
+}
+
+static void toggle_dac(Section* section)
```

Comment:
Yeah... I don't like "dual-purposing" `section` so it can either be a kind of a flag to just disable the DAC (when it's null), or an actual section that holds the config value, then based on the config it will be either disabled or enabled.

The function is doing a bit too much. Could it be broken down into separate parts so it can be composed in a less tangled way? E.g. one function to parse the config settings and come up with a decision whether to enable or disable the DAC, then a second one to disable the DAC, and a third one to enable it.

`toggle_dac` has two invocations: one always disables the DAC, the other invocation either enables or disables it based on the config. So with three-function approach, the intent would be clearer and a lot more explicit — no more passing of "magic null sections" that in fact acts as "disable DAC":

- 1st invocation always calls `disable_dac`
- 2nd invocation calls `is_dac_enabled`, then based on the result it calls either `enable_dac` or `disable_dac`.

The less things are "entwined", the better, IMO 😄